### PR TITLE
Utxo package refactoring

### PR DESCRIPTION
**Pull request by i-skrypnyk**

### CODE REVIEW

**Comment by i-skrypnyk** [comment]
It seems that for the short files, like <10-15 lines, idea and git don't want to treat it like renaming and always say it's delete-create changes

**Review by i-Alex — CHANGES_REQUESTED** [review]
Check the following places:
1. ApplicationNodeViewProvider - seems no one is using -> can be removed at all.
2. NodeViewHelperImpl and TransactionSubmitHelperImpl are Scala class in `Java` part. Do we need them to be written in Scala? If yes move to `Scala` part. If not -> make Java and do the same for Account specific files: `AccountTransactionSubmitHelperImpl`, `AccountNodeViewHelperImpl`
3. Check the package of `MainchainBlockReferenceInfo` and `MainchainBlockReferenceInfoSerializer`. I think we must put it `com.horizen.chain`
4. review `serialization` package. I think we need to call it `json` and inside `json.serializer`  (for Serializers only; `json mapper` can be put into `json` package directly).
5. `com.horizen.account.events` -> `com.horizen.account.state.events` (both scala and java parts)
6.  `com.horizen.account.receipt` -> `com.horizen.account.state.receipt` all except `Bloom` (it can be moved to `account.utils`
7. `account.state` package seems a bit messy, we may split it in several sub packages (to be discussed).
8. `api.http` let's move all routes to a `api.http.route` package. It will help a lot especially in a common package.
9. `RpcIdSerializer` move to account
10. review `com.horizen.utils`: move CswData, FeePaymentsUtils, MempoolMap, UtxoMerkleTreeLeafInfo into `utxo` package.
11. utxo and common package put into a separate subpackages state, wallet, mempool, history; like it is currently structured in `account` model
12. after point (11) move `validation` as a subpackage of `history`

File: sdk/src/test/scala/com/horizen/account/block/AccountBlockTest.scala [hunk 1]
```diff
@@ -23,6 +23,7 @@ import com.horizen.proposition.VrfPublicKey
 import com.horizen.secret.VrfSecretKey
 import com.horizen.serialization.{ApplicationJsonSerializer, SerializationUtil}
 import com.horizen.utils.{BytesUtils, TestSidechainsVersionsManager}
+import com.horizen.utxo.block.SidechainBlock
```
[the remark this answers is not in the document]
Comment (i-skrypnyk) [reply sdk/src/test/scala/com/horizen/account/block/AccountBlockTest.scala]: This is a specific negative check for a wrong version from utxo model

[the remark this answers is not in the document]
Comment (i-Alex) [reply sdk/src/test/scala/com/horizen/account/block/AccountBlockTest.scala]: ok, got it.

**Review by i-Alex — CHANGES_REQUESTED** [review]
New batch of changes seems good and addressing previous comments.
Beside that a couple of other things:
1. MempoolMap move to mempool package
2. Move SchnorrFunctions into `package com.horizen.cryptolibprovider`
3. couple of places where i see `receipt.EthereumReceipt` -> can be just `EthereumReceipt`
4. SidechainBackup can be moved to `com.horizen.utxo.state`

**Comment by i-skrypnyk** [comment]
Applied the remaining fixes

**Review by i-Alex — APPROVED** [review]

File: sdk/src/main/scala/com/horizen/utxo/SidechainStateUtxoMerkleTreeProvider.scala [hunk 2]
```diff
@@ -1,9 +1,11 @@
-package com.horizen
+package com.horizen.utxo
 
-import com.horizen.box.CoinsBox
+import com.horizen.SidechainTypes
```
[the remark this answers is not in the document]
Comment (i-Alex) [reply sdk/src/main/scala/com/horizen/utxo/SidechainStateUtxoMerkleTreeProvider.scala]: will add to the backlog

**Pull request merged**